pkg/storage/store.go

```diff
@@ -1118,9 +1117,19 @@ func (s *Store) Start(ctx context.Context, stopper *stop.Stopper) error {
 			if err != nil {
 				return false, err
 			}
-			if err = s.addReplicaInternalLocked(rep); err != nil {
-				return false, err
+
+			// We can't lock s.mu across NewReplica due to the lock ordering
+			// constraint (*Replica).raftMu < (*Store).mu. See the comment on
+			// (Store).mu.
+			{
+				s.mu.Lock()
+				if err = s.addReplicaInternalLocked(rep); err != nil {
+					s.mu.Unlock()
+					return false, err
+				}
+				s.mu.Unlock()
```

Comment:
I'd write this as:

```
s.mu.Lock()
err := s.addReplicaInternalLocked(rep)
s.mu.Unlock()
if err != nil {
  return false, err
}
```

Do you know if there was something in the code that required holding `Store.mu` during the replica iteration?

Reply:
Ah, good point. Done.

I checked the code and past diffs and I don't think anything else needs the lock.